### DISCUSSION TITLE
Possibly batch_resampler related issue in r1.16.0 VITS TTS model trainer

### DISCUSSION THREAD

**Discussion by oytunturk** [NVIDIA-NeMo/NeMo#6235] [Q&A]
Hi,

I get an error when trying to train a VITS TTS model:

`
.../NeMo/nemo/collections/tts/torch/data.py", line 1540, in __iter__
    ids_bucket = ids_bucket + ids_bucket * (rem // len_bucket) + ids_bucket[: (rem % len_bucket)]
ZeroDivisionError: integer division or modulo by zero`

It seems 'len_bucket' sometimes becomes zero for some reason. Is there a way to adjust config parameters to get around this issue? Maybe 'batch_size' and/or 'boundaries' in the vits config file?

Thanks!

**Comment by XuesongYang** [comment]
@treacker could you pls have a look?

**Reply by oytunturk** [reply]
Update after trying a few things: It happens when fft-size is set to 1024 at 22KHz sampling rate. 2048 point fft works but it’s probably unnecessary resolution at that rate.
